lib/helpers/fields/statistics-fields.js

```diff
@@ -0,0 +1,102 @@
+const _ = require('lodash')
+const { isDate } = require('./data-type-utils')
+const {
+  StatisticField,
+  StatisticDateField,
+  FieldFromFieldDefinition,
+  FieldFromKeyValue
+} = require('./field-classes')
+
+class StatisticsFields {
+  static normalizeOptions (inputOptions) {
+    const {
+      statistics,
+      metadata: {
+        fields
+      } = {},
+      groupByFieldsForStatistics = [],
+      attributeSample,
+      ...options
+    } = inputOptions
+
+    return {
+      statisticsSample: Array.isArray(statistics) ? statistics[0] : statistics,
+      fieldDefinitions: options.fieldDefinitions || options.fields || fields,
+      groupByFieldsForStatistics: Array.isArray(groupByFieldsForStatistics) ? groupByFieldsForStatistics : groupByFieldsForStatistics
+        .replace(/\s*,\s*/g, ',')
+        .replace(/^\s/, '')
+        .replace(/\s*$/, '')
+        .split(','),
+      ...options
+    }
+  }
+
+  static create (inputOptions) {
+    const options = StatisticsFields.normalizeOptions(inputOptions)
+    return new StatisticsFields(options)
+  }
+
+  constructor (options = {}) {
+    const {
+      statisticsSample,
+      groupByFieldsForStatistics = [],
+      fieldDefinitions = [],
+      outStatistics
+    } = options
+    const dateFieldRegexs = getDateFieldRegexs(fieldDefinitions, outStatistics)
+
+    return Object
+      .entries(statisticsSample)
+      .map(([key, value]) => {
+        if (groupByFieldsForStatistics.includes(key)) {
+          const fieldDefinition = fieldDefinitions.find(({ name }) => name === key)
+
+          if (fieldDefinition) {
+            return new FieldFromFieldDefinition(fieldDefinition)
+          }
+
+          return new FieldFromKeyValue(key, value)
+        }
+
+        if (isDateField(dateFieldRegexs, key, value)) {
+          return new StatisticDateField(key)
+        }
+
+        return new StatisticField(key)
+      })
+  }
```

Comment:
Can all this be wrapped in a helper?

Reply:
It can be - are you suggesting that it should be?

Reply:
There really isn't much going on in the constructor, so I didn't think moving this to its function really help maintain the current level of abstraction.  But ready to hear your case!